modules/shared/components/HOC/WithErrorHandler/WithErrorHandler.test.tsx

```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import WithErrorHandler from './WithErrorHandler';
+import type { IDataTests } from './IDataTests';
+
```

Comment:
I believe one test that renders the same component twice (one with success data and another with error) should be the way to test this HoC.